documentation/source/task_bfms.rst

```diff
@@ -0,0 +1,196 @@
+            end
+         end
+         
+         always @(posedge clock) begin
+            if (reset) begin
+               data_valid <= 0;
+               data <= 0;
+            end else begin
+               data_valid <= data_valid_v;
+               data <= data_v;
+               if (data_valid && data_ready) begin
+                  write_ack();
+                  data_valid_v = 0;
+               end
+            end
+         end
+         
+         task write_req(reg[63:0] d);
+            begin
+               data_v = d;
+               data_valid_v = 1;
+            end
+         endtask
+      
+         // Auto-generated code to implement the BFM API
+      ${cocotb_bfm_api_impl}
+      
+      endmodule
+      
+The implementation of ``export`` tasks (Python methods called from HDL) 
+and the machinery to call ``import`` tasks is substituted into the
+template via where the ``${cocotb_bfm_api_impl}`` macro is referenced.
+
+Using BFMs from HDL
+===================
+The HDL portion of the testbench must instantiate BFMs where needed.
+These instances will register their existence with Cocotb when simulation
+starts. 
+
+
+Using BFMs from Python
+======================
+Available BFM instances are registered with the :class:`cocotb.BfmMgr` class. 
+Static methods provide access to the list of available BFMs, and the
+:meth:`~cocotb.BfmMgr.find_bfm` method accepts a regular expression to find a BFM based
+on its HDL instance path.
+
+.. code-block:: python3
+
+    @cocotb.coroutine
+    def run_c(self):
+        out_bfm = BfmMgr.find_bfm(".*u_bfm")
+        
+        for i in range(1,101):
+            yield out_bfm.write_c(i)
+
+The code snippet above shows typical use within a test. The :meth:`~cocotb.BfmMgr.find_bfm`
+method is used to find a BFM with the expected instance path. Then,
+methods on the BFM object are called to send data via the BFM.
+
+Cocotb Makefile Interface
+=========================
+If you are using the Cocotb Makefiles, simply append the BFM packages
+used by your testbench to the :make:var:`COCOTB_BFM_MODULES` variable
+
+.. code-block:: make
+
+    COCOTB_BFM_MODULES += rv_bfms
+    
+The Makefiles will automatically generate and compile the interface
+files along with the rest of your testbench.
+
+Manually Generating BFM Interface Files
+=======================================
+The interface code that allows Cocotb to call HDL tasks, and to enable
+HDL to call Python methods is auto-generated. This ensures that the 
+HDL interface is always up-to-date with the Python definition of the
+BFM API.
+
+The :command:`cocotb-bfmgen` script generates the appropriate BFM interface
+files based on the BFMs required for a given testbench.
+
+The ``cocotb-bfmgen`` script accepts the following options:
+
+- -m <module> -- Specifies a Python module to load. Typically, this will
+  be a BFM package.
+- -l,--language <target> -- Specifies the target testbench language. ``vlog`` and ``sv`` 
```

Comment:
```suggestion
- ``-l``, ``--language <target>`` -- Specifies the target testbench language. ``vlog`` and ``sv`` 
```
otherwise Sphinx will render the ``--`` as an em dash (https://external-builds.readthedocs.io/html/cocotb/1217/task_bfms.html#manually-generating-bfm-interface-files)
The other options should get the same treatment if only for consistency.